lectures/12-plonk.tex

```diff
@@ -0,0 +1,234 @@
+\documentclass[../lecture-notes.tex]{subfiles}
+\usepackage[T1]{fontenc}
+\usepackage{lmodern}
+\usepackage{xcolor}
+\usepackage{fix-cm}
+\pgfplotsset{compat=1.18}
+
+\begin{document}
+
+\subsection{Preliminaries}
+
+Before delving into Plonk, let's first consider a few useful proving system gadgets.
+
+Let $\Omega$ be some subset of $\mathbb{F}_p$ of size $k$.
+Let $f \in \mathbb{F}_p^{(\leq d)}[X] \qquad (d \geq k)$ \hfill \begin{tabular}{|c|}\hline Verifier has $f$ \\ \hline \end{tabular} 
+
+Now we need to construct efficient Poly-IOPs for two tasks described in below sections.
+
+\subsubsection{ZeroTest}
+
+Prove that $f$ is identically zero on $\Omega$.
+
+\begin{enumerate}
+    \item Prover computes: \( q(X) \leftarrow \frac{f(X)}{Z_{\Omega}(X)} \quad q \in \mathbb{F}_p^{(\leq d)}[X] \)
+    \item Prover sends commitment \(\left[ q \right]\) to the verifier
+    \item Verifier generates random \(r \leftarrow\mathbb{F}_p^{(\leq d)} \)
+    \item Verifier queries from the prover \(q(X)\) and \(f(X)\) at \(r\)
+    \item Verifier accepts if $f(r) = q(r) \cdot Z_{\Omega}(r)$ (implies that $f(X) = q(X) \cdot Z_{\Omega}(X)$ with high probability).
+\end{enumerate}
+
+\begin{lemma}
+    $f$ is zero on $\Omega$ if and only if $f(X)$ is divisible by $Z_{\Omega}(X)$.
+\end{lemma}
+
+\begin{remark}
+This protocol is complete and sound, assuming $d/p$ is negligible.
+\end{remark}
+
+\subsubsection{Prescribed Permutation Check}
+
+\begin{definition}
+\(W : \Omega \rightarrow \Omega\) is a permutation of \(\Omega\) if \(\forall i \in [k], W(\omega^i) = \omega^{j}\) is a bijection.
+\end{definition}
+
+\begin{example}
+\text{Example } $(k=3): \quad W(\omega^0) = \omega^2, \quad W(\omega^1) = \omega^0, \quad W(\omega^2) = \omega^1$
+\end{example}
+
+Let \(f\) and \(g\) be polynomials in \(\mathbb{F}_p^{(\leq d)}[X]\). Verifier has commitments \([f]\), \([g]\), \([W]\).
```

Comment:
Specify that you are using notation $[\star]$ to denote that we have a commitment to $\star$.

Reply:
Or even better, use notation we used in the previous lectures: $\mathsf{com}(f)$, ...